content/ADR-148-processing-scene-messages.md

```diff
@@ -0,0 +1,113 @@
+
+## Context, Reach & Prioritization
+
+<!--
+Discuss and go into detail about the subject in question. Make sure you cover:
+- Why is this decision important
+- The urgency of the decision
+- Datapoints and related background information
+- Vocabulary and key terms
+-->
+
+Decentraland scenes run in contexts isolated from the rendering engine (_renderer_ from now on), in a worst case scenario, in a different process only being able to communicate via messaging. Since scenes have an independent update loop (_scene frame_ from now on) clear synchronization points need to be designed to reach consistent states between the renderer and the scene.
+
+The context of synchronizing the scenes and the renderer is complex. Many dimensions participate in the analysis:
+- Dropping scene frames due to waits or locks
+- Input delay (time or frames between the interaction and effect)
+- Optimization of compute time (or what can be executed while the main thread is waiting for the GPU)
+
+
+## Solution Space Exploration
+
+This document will enumerate the considered alternatives and the implications of each one of them.
+
+### Naive approach
+
+This approach is the simplest to explain: A scene frame runs in the Scene, then it sends the updates to the renderer. It waits for the response to run the next scene frame.
+
+<figure>
+  <img src="/resources/ADR-148/Frame 1.svg" />
+</figure>
+
+#### Implications
+
+- The scene will run at half the speed of the renderer in the best case sceneario, introducing Dropped frames.
+- The renderer will maintain full speed in the rendering, but every other frame will be the same as the previous one.
+- If the update of the scene reaches the renderer mid-frame, then two frames will be dropped for the scene.
+
+### Extension
+
+To illustrate better what happens inside the scene frame and renderer frame, we will consider some stages for each of them:
+
+For the scene frame:
+- **Scene.Recieve**: Read all the messages from the renderer
+- **Scene.Update**: Execute scene logic, run systems, mutate state
+- **Scene.Send**: Send all the updates to the renderer
+
+For the renderer frame:
+- **Renderer.Recieve**: Read all the messages from the scenes
+- **Renderer.Update**: Update transforms, calculate physics, etc.
+- **Renderer.Render**: Batch GPU commands and send them to the GPU Process
+- **Renderer.Send**: Send all the updates to the scene
+
+<figure>
+  <img src="/resources/ADR-148/Frame 2.svg" />
+</figure>
+
+### Proposed approach
+
+Now that the frame is decomposed into smaller chunks, it can be observed that the RENDER part doesn't necessarily needs to halt the **Renderer.Send** back to the scene, but it is a requirement for the **Renderer.Render**, since physics and transformations are used to calculate the GPU buffers for the next frame.
```

Comment:
```suggestion
Now that the frame is decomposed into smaller chunks, it can be observed that the RENDER part doesn't necessarily need to halt the **Renderer.Send** back to the scene, but it is a requirement for the **Renderer.Render**, since physics and transformations are used to calculate the GPU buffers for the next frame.
```